lib/src/chart/line_chart/line_chart_painter.dart

```diff
@@ -1378,13 +1378,37 @@ class LineChartPainter extends AxisChartPainter<LineChartData> {
     /// draw the texts one by one in below of each other
     var topPosSeek = tooltipData.tooltipPadding.top;
     for (var tp in drawingTextPainters) {
-      final drawOffset = Offset(
-        rect.center.dx - (tp.width / 2),
-        rect.topCenter.dy +
-            topPosSeek -
-            textRotationOffset.dy +
-            rectRotationOffset.dy,
-      );
+      double yOffset = rect.topCenter.dy +
+          topPosSeek -
+          textRotationOffset.dy +
+          rectRotationOffset.dy;
+
+      final ui.Offset drawOffset;
+
+      if ((tp.textAlign == TextAlign.left) ||
```

Comment:
Can you please make this logic more readable? I mean by making some functions and calling them here.

Reply:
We can make an extension function like below:
```dart
enum HorizontalAlignment {
  left, center, right
}

extension TextAlignExtension on TextAlign {
  HorizontalAlignment getFinalHorizontalAlignment(TextDirection? direction) {
    if (((this == TextAlign.left) ||
        (this == TextAlign.start && direction == TextDirection.ltr) ||
        (this == TextAlign.end && direction == TextDirection.rtl))) {
      return HorizontalAlignment.left;
    } else if ((this == TextAlign.right) ||
        (this == TextAlign.end && direction == TextDirection.ltr) ||
        (this == TextAlign.start && direction == TextDirection.rtl)) {
      return HorizontalAlignment.right;
    } else {
      return HorizontalAlignment.center;
    }
  }
}
```

Then we can use it in our code in the painter. (It helps us to reuse it later on other classes)

Reply:
@imaNNeoFighT  I think we can keep it in `extensions` folder, my bad I didn't see clearly. What file name though?

Reply:
Yes, we need to put it into the `extensions/` directory.
Let's call it`text_align_extension.dart`. 
Sounds good?

Reply:
done.